cns/ipampool/metrics.go

```diff
@@ -132,4 +140,10 @@ func observeIPPoolState(state ipPoolState, meta metaState, labels []string) {
 	ipamPrimaryIPCount.WithLabelValues(labels...).Set(float64(len(meta.primaryIPAddresses)))
 	ipamRequestedIPConfigCount.WithLabelValues(labels...).Set(float64(state.requestedIPs))
 	ipamTotalIPCount.WithLabelValues(labels...).Set(float64(state.totalIPs))
+	if meta.exhausted {
+		ipamSubnetExhaustionState.WithLabelValues(labels...).Set(float64(1))
+	} else {
+		ipamSubnetExhaustionState.WithLabelValues(labels...).Set(float64(0))
+	}
+
```

Comment:
it's more common in go to start with the base case, so that there is no need of an `else` branch at all. 
```suggestion
	var exhaustionState float64 // = 0 if you want to be really clear 
	if meta.exhausted {
		exhaustionState = 1
	}
	ipamSubnetExhaustionState.WithLabelValues(labels...).Set(exhaustionState)
```
here it's minor, but in cases where the branching logic is very complex, this is much more readable.

Reply:
Thanks for the suggestion. Changed